src/main/c/VipsImage.c

```diff
@@ -164,6 +164,18 @@ Java_com_criteo_vips_VipsImageImpl_resizeNative(JNIEnv *env, jobject obj, jint w
     g_object_unref(im);
 }
 
+JNIEXPORT void JNICALL
+Java_com_criteo_vips_VipsImageImpl_autoRotateNative(JNIEnv *env, jobject obj)
+{
+    VipsImage *im = (VipsImage *) (*env)->GetLongField(env, obj, handle_fid);
+    VipsImage *out = NULL;
+
+    vips_autorot(im, &out, NULL);
```

Comment:
Can you please check the vips_autorot() returned value and throw a VipsException in case of failure (-1)?